packages/lesswrong/server/repos/PostsRepo.ts

```diff
@@ -901,6 +901,23 @@ class PostsRepo extends AbstractRepo<"Posts"> {
       maxAgeDays: `${maxAgeDays} days`,
     });
   }
+  
+  async ensurePostHasNonDraftContents(postId: string) {
+    await this.none(`
+      UPDATE "Revisions" AS r
+      SET
+        "draft" = FALSE,
+        "version" = CASE WHEN LEFT("version", 1) = '0' THEN '1.0.0' ELSE "version" END
+      FROM "Posts" AS p
+      WHERE
+        p._id = $(postId)
+        AND p."contents_latest" = r."_id"
+        AND p."draft" IS NOT TRUE
+        AND r."draft" IS TRUE
```

Comment:
```suggestion
```

I think we need to remove the `r."draft" IS TRUE` to ensure `version` is updated correctly if draft is already false but the version is less than 1.

Reply:
I think that case won't ever happen (and if it did happen it would imply something else weird going on, such that the desired behavior wouldn't be obvious).